packages/build-tools/src/builders/android.ts

```diff
@@ -11,7 +11,32 @@ import { restoreCredentials } from '../android/credentials';
 import { configureBuildGradle } from '../android/gradleConfig';
 import { prebuildAsync } from '../utils/prebuild';
 
-export default async function androidBuilder(ctx: BuildContext<Android.Job>): Promise<string[]> {
+export default async function androidBuilder(ctx: BuildContext<Android.Job>): Promise<string> {
+  let isBuildSuccess = true;
+  try {
+    const archiveLocation = await buildAsync(ctx);
+    await ctx.runBuildPhase(BuildPhase.ON_BUILD_SUCCESS_HOOK, async () => {
+      await runHookIfPresent(ctx, Hook.ON_BUILD_SUCCESS);
+    });
+    return archiveLocation;
+  } catch (err: any) {
+    isBuildSuccess = false;
+    await ctx.runBuildPhase(BuildPhase.ON_BUILD_ERROR_HOOK, async () => {
+      await runHookIfPresent(ctx, Hook.ON_BUILD_ERROR);
+    });
+    throw err;
+  } finally {
+    await ctx.runBuildPhase(BuildPhase.ON_BUILD_COMPLETED_HOOK, async () => {
+      await runHookIfPresent(ctx, Hook.ON_BUILD_COMPLETED, {
+        extraEnvs: {
+          EAS_BUILD_STATUS: isBuildSuccess ? 'success' : 'error',
```

Comment:
I know that "errored" is not correct but I think we're using it everywhere.
```suggestion
          EAS_BUILD_STATUS: isBuildSuccess ? 'success' : 'errored',
```

Reply:
if that is the case, we should change success to finished or succeeded. The problem is that with finished is not obvious that build was finished correctly, and succeeded sounds a bit weird

Reply:
It should match the status sent to build webhooks.